hedera-node/src/main/java/com/hedera/services/state/virtual/UniqueTokenKey.java

```diff
@@ -0,0 +1,152 @@
+package com.hedera.services.state.virtual;
+
+import com.swirlds.common.io.SerializableDataInputStream;
+import com.swirlds.common.io.SerializableDataOutputStream;
+import com.swirlds.virtualmap.VirtualKey;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+/**
+ * Represents a key for a unique token (NFT).
+ */
+public class UniqueTokenKey implements VirtualKey<UniqueTokenKey> {
+	private static final long CLASS_ID = 0x17f77b311f6L;
+
+	/** Current version of the encoding scheme. */
+	/* package */ static final int CURRENT_VERSION = 1;
+
+	/**
+	 * Expected maximum number of bytes this class will serialize to. Serialization will format will be:
+	 *  - (1 byte) number of bytes, 1-255, following this byte that will contain the token number.
+	 *  - (variable, up to 8 bytes)  the non-leading zero-bytes representing the token number.
+	 */
+	public static final int ESTIMATED_SIZE_BYTES = Long.BYTES + 1;
+
+
+	/** The token number from the address will be used as the key.*/
+	private long tokenNum;
+	/** Hashcode will be updated whenever tokenNum changes.*/
+	private int hashCode;
+
+	public UniqueTokenKey() {}
+
+	public UniqueTokenKey(long tokenNum) {
+		setTokenNum(tokenNum);
+	}
+
+	public long getTokenNum() {
+		return tokenNum;
+	}
+
+	private void setTokenNum(long tokenNum) {
+		this.tokenNum = tokenNum;
+		this.hashCode = Long.hashCode(tokenNum);
```

Comment:
On a recent vacation 😁, Leemon found a [near-optimal hash function](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/utils/MiscUtils.java#L798) for a `long`---we use it e.g. [here](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/utils/EntityNum.java#L167).